File: online_judge/hackerrank/plus_minus.cpp
#include <cstdio>

int n;
int p = 0;
int ne = 0;
int z = 0;
int input = 0;

int main(){

    scanf("%d", &n);
    for(int i = 0; i < n; ++i){
        scanf("%d", &input);
        if(input == 0) ++z;
        else if(input > 0) ++p;
        else ++ne;
    }
    printf("%.6lf\n", 1.0*p/n);
    printf("%.6lf\n", 1.0*ne/n);
    printf("%.6lf\n", 1.0*z/n);

    return 0;
}

File: online_judge/hackerrank/plus_minus.py
import sys

n = int(input().strip())

arr = [int(x) for x in input().strip().split(' ')]

cnt_pos = 0
cnt_neg = 0
cnt_zero = 0
cnt = len(arr)


for x in arr:
    if x > 0:
        cnt_pos += 1
    elif x == 0:
        cnt_zero += 1
    else:
        cnt_neg += 1

print("%.5f" % (cnt_pos / cnt))
print("%.5f" % (cnt_neg / cnt))
print("%.5f" % (cnt_zero / cnt))
